src/sinks/datadog/logs/sink.rs

```diff
@@ -384,56 +390,42 @@ where
         let partitioner = EventPartitioner::default();
 
         let builder_limit = NonZeroUsize::new(64);
+        let input = input.batched_partitioned(partitioner, self.batch_settings);
         if self.schema_enabled {
-            let sink = input
-                .batched_partitioned(partitioner, self.batch_settings)
-                .request_builder(
-                    builder_limit,
-                    SemanticLogRequestBuilder {
-                        default_api_key,
-                        encoding: SemanticJsonEncoding {
-                            log_schema: self.encoding.log_schema,
-                            encoder: self.encoding.encoder,
-                        },
-                        compression: self.compression,
+            input.request_builder(
+                builder_limit,
+                SemanticLogRequestBuilder {
+                    default_api_key,
+                    encoding: SemanticJsonEncoding {
+                        log_schema: self.encoding.log_schema,
+                        encoder: self.encoding.encoder,
                     },
-                )
-                .filter_map(|request| async move {
-                    match request {
-                        Err(error) => {
-                            emit!(SinkRequestBuildError { error });
-                            None
-                        }
```

Comment:
It's not apparent to me why this code was deleted :thinking:

Reply:
Both branches of the `if/else` had the same prelude (`batched_partitioned`) and postlude (`.filter_map.into_driver`) so I merged them together outside of the conditional part. It appears in the diff below.